com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -317,6 +330,10 @@ internal struct AgentParameters
         /// </summary>
         float[] m_LegacyHeuristicCache;
 
+        int m_GroupId;
+
+        internal event Action<Agent> UnregisterFromGroup;
```

Comment:
This is the delegate for agent to trigger removing itself from the group and is added to avoid cyclic reference between agent and group. The agent needs a way to clean itself up from the group when disabled/destroyed/assigned to another group.

I'll add comments.

Reply:
This might be better with a more generic name, like OnAgentDisabled.

Reply:
didn't named it OnAgentDisabled since it was also triggered in SetMultiAgentGroup. Now that is removed I changed the name.